### ISSUE TITLE
[BUG] `_scan_profiles` strips `h` and `m` from profile name

### ISSUE THREAD

**Issue by ezherman** [gem-pasteur/macsyfinder#69]
**Describe the bug**
The `_scan_profiles` function removes the `.hmm` extension using `strip`. However, this also removes trailing `h` and `m` characters from profile names, resulting in Macsyfinder being unable to find a profile.

**To Reproduce**
Please see https://github.com/mdmparis/defense-finder/issues/42, where I outlined the issue and @ZarulHanifah reported that it is driven by `strip`. 

**Expected behavior**
The command should exclusively remove the file extension. One option would be to use `replace`, i.e. 

```
if os.path.isfile(profile_path):
    if profile.endswith(profile_suffix):
        base = profile.replace(profile_suffix, '')
    elif profile.endswith(compressed_suffix):
        base = profile.replace(compressed_suffix, '')
```

**Please complete the following information):**

***OS:*** 

 - [x] Linux
 - [ ] Windows
 - [ ] Mac

***MacSyFinder Version:***
```
Macsyfinder 2.1.3
using:
- Python 3.12.1 | packaged by conda-forge | (main, Dec 23 2023, 08:03:24) [GCC 12.3.0]
- NetworkX 3.2.1
- Pandas 2.1.4
```

**Comment by jewell-bug** [comment]
I was wondering if this bug fix will be incorporated in a conda release in the near future. Or alternatively is there a previous version we can specify that does not have this bug? Thank you

**Comment by alexweisberg** [comment]
Seconding Jewell's comment, is there any chance that a new bump release of MacSyFinder containing this fix commit could be released? Myself and others install macsyfinder and defensefinder via conda and this bug is preventing it from running. Thanks!

**Comment by jeanrjc** [comment]
We released defensefinder (v1.2.2) with a previous version of macsyfinder that did not have the bug. We'll make another release when macsyfinder will be released.
